app/scripts/migrations/097.ts

```diff
@@ -0,0 +1,49 @@
+import { cloneDeep, isEmpty } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 97;
+
+/**
+ * Remove `nonceDetail` from transactions
+ *
+ * @param originalVersionedData
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, any>) {
+  const TransactionController = state?.TransactionController || {};
+  const transactions = state?.TransactionController?.transactions || {};
+
+  if (isEmpty(TransactionController) || isEmpty(transactions)) {
```

Comment:
As have a fallback on the full transaction controller state, could we just rely on the second condition as it will be empty if the controller state is empty?

Reply:
Done

Reply:
Done